db/error_handler.cc

```diff
@@ -664,11 +664,17 @@ const Status& ErrorHandler::StartRecoverFromRetryableBGIOError(
     // wait the previous recover thread to finish and create a new thread
     // to recover from the bg error.
     db_mutex_->Unlock();
+    TEST_SYNC_POINT(
+        "StartRecoverFromRetryableBGIOError:BeforeWaitingForOtherThread");
     old_recovery_thread->join();
+    TEST_SYNC_POINT(
+        "StartRecoverFromRetryableBGIOError:AfterWaitingForOtherThread");
     db_mutex_->Lock();
+
+    // In case recovery_in_prog_ was set back to false by old_recovery_thread
+    recovery_in_prog_ = true;
```

Comment:
The old recovery thread could also clear the bg error when setting recovery_in_prog_ to false. I think we could be doing recovery work even though bg_error_ is OK and background work is not stopped (soft_error_no_bg_work_ is also cleared). I wonder if we should just clear bg error and relevant states in error handler and not in ResumeImpl()